src/api/cpp/GravityNode.cpp

```diff
@@ -568,641 +565,646 @@ GravityNode::~GravityNode()
+
+        // Configure to trap Ctrl-C (SIGINT) and SIGTERM signals
+        s_catch_signals();
+
+        // wait for the manager threads to signal their readiness
+        string msgText;
+        int numThreadsWaiting = 5;
+        while (numThreadsWaiting && !s_interrupted)
+        {
+            // Read message
+            msgText = readStringMessage(initSocket);
+
+            // respond with ack
+            sendStringMessage(initSocket, "ack", ZMQ_DONTWAIT);
+
+            // Decrement counter
+            numThreadsWaiting--;
+        }
+        zmq_close(initSocket);
+
+        s_restore_signals();
+
+        if (s_interrupted)
+        {
+            initLock.Unlock();
+            raise(s_interrupted);
+        }
+
+        // connect down here to make sure manager has bound address.
+        publishManagerRequestSWL.socket = zmq_socket(context, ZMQ_REQ);
+        zmq_connect(publishManagerRequestSWL.socket, PUB_MGR_REQ_URL);
+
+        serviceManagerSWL.socket = zmq_socket(context, ZMQ_REQ);
+        zmq_connect(serviceManagerSWL.socket, SERVICE_MGR_URL);
+
+        // Configure high water marks
+        int publishHWM = getIntParam("PublishHWM", 1000);
+        if (publishHWM < 0)
+        {
+            logger->warn("Invalid PublishHWM = {}. Ignoring.", publishHWM);
+        }
+        else
+        {
+            // Send HWM (REQ/REP)
+            sendStringMessage(publishManagerRequestSWL.socket, "set_hwm", ZMQ_SNDMORE);
+            sendIntMessage(publishManagerRequestSWL.socket, publishHWM, ZMQ_DONTWAIT);
+            // Read ACK
+            readStringMessage(publishManagerRequestSWL.socket);
+        }
+        int subscribeHWM = getIntParam("SubscribeHWM", 1000);
+        if (subscribeHWM < 0)
+        {
+            logger->warn("Invalid subscribeHWM = {}. Ignoring.", subscribeHWM);
+        }
+        else
+        {
+            // Send HWM (PUB/SUB)
+            sendStringMessage(subscriptionManagerSWL.socket, "set_hwm", ZMQ_SNDMORE);
+            sendIntMessage(subscriptionManagerSWL.socket, subscribeHWM, ZMQ_DONTWAIT);
+        }
+
+        // Configure TCP keep-alive
+        if (getBoolParam("TcpKeepAliveEnabled", false))
+        {
+            // Get settings
+            int tcpKeepAliveTime = getIntParam("TcpKeepAliveTime", 7200);
+            int tcpKeepAliveProbes = getIntParam("TcpKeepAliveProbes", 9);
+            int tcpKeepAliveIntvl = getIntParam("TcpKeepAliveIntvl", 75);
+
+            logger->info("Enabling TCP publisher Keep-Alive with settings (time={}, probes={}, intvl={}).",
+                         tcpKeepAliveTime, tcpKeepAliveProbes, tcpKeepAliveIntvl);
+
+            // Send TCP keep-alive settings
+            sendStringMessage(publishManagerRequestSWL.socket, "set_tcp_keepalive", ZMQ_SNDMORE);
+            sendIntMessage(publishManagerRequestSWL.socket, tcpKeepAliveTime, ZMQ_DONTWAIT);
+            sendIntMessage(publishManagerRequestSWL.socket, tcpKeepAliveProbes, ZMQ_DONTWAIT);
```

Comment:
Not sure why the conflict was so large, but these 2 lines are all that are needed from the last PR that was merged into master.
```suggestion
            sendIntMessage(publishManagerRequestSWL.socket, tcpKeepAliveTime, ZMQ_SNDMORE);
            sendIntMessage(publishManagerRequestSWL.socket, tcpKeepAliveProbes, ZMQ_SNDMORE);
```